src/main/rx/ghst.c

```diff
@@ -212,40 +215,58 @@ static bool ghstProcessFrame(const rxRuntimeState_t *rxRuntimeState)
     }
 
     if (ghstValidatedFrameAvailable) {
-        int startIdx = 4;
-        switch (ghstValidatedFrame.frame.type) {
-            case GHST_UL_RC_CHANS_HS4_5TO8:
-            case GHST_UL_RC_CHANS_HS4_9TO12:
-            case GHST_UL_RC_CHANS_HS4_13TO16: {
-                const ghstPayloadPulses_t* const rcChannels = (ghstPayloadPulses_t*)&ghstValidatedFrame.frame.payload;
-
-                // all uplink frames contain CH1..4 data (12 bit)
-                ghstChannelData[0] = rcChannels->ch1 >> 1;
-                ghstChannelData[1] = rcChannels->ch2 >> 1;
-                ghstChannelData[2] = rcChannels->ch3 >> 1;
-                ghstChannelData[3] = rcChannels->ch4 >> 1;
+        int startIdx = 0;
+
+        if(ghstValidatedFrame.frame.type >= GHST_UL_RC_CHANS_HS4_FIRST && 
+           ghstValidatedFrame.frame.type <= GHST_UL_RC_CHANS_HS4_LAST) {
+            const ghstPayloadPulses_t* const rcChannels = (ghstPayloadPulses_t*)&ghstValidatedFrame.frame.payload;
+
+            // all uplink frames contain CH1..4 data (12 bit)
+            ghstChannelData[0] = rcChannels->ch1to4.ch1 >> 1;
+            ghstChannelData[1] = rcChannels->ch1to4.ch2 >> 1;
+            ghstChannelData[2] = rcChannels->ch1to4.ch3 >> 1;
+            ghstChannelData[3] = rcChannels->ch1to4.ch4 >> 1;
+
+            switch(ghstValidatedFrame.frame.type) {
+                case GHST_UL_RC_CHANS_HS4_RSSI: {
```

Comment:
```suggestion
                case GHST_UL_RC_CHANS_HS4_RSSI:
```

Reply:
Hum... compiler doesn't like those being removed. The declaration of rssiFrame immediately follows the case, which as far as the compiler is concerned is a label: 

https://www.educative.io/edpresso/resolving-the-a-label-can-only-be-part-of-a-statement-error

Reply:
Thanks. Learned something new.